storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -4214,6 +4203,31 @@ private void maybeAddPlaceholderBoltAggStats(TopologyPageInfo topoPageInfo, Stor
         }
     }
 
+    private CommonAggregateStats getPlaceholderCommonAggregateStats(Object component) {
+        // common aggregate
+        CommonAggregateStats commonStats = new CommonAggregateStats();
+
+        // get num_executors
+        int numExecutors = 0;
+        try {
+            numExecutors = StormCommon.numStartExecutors(component);
+        } catch (InvalidTopologyException e) {
+            // ignore
+        }
+
+        // get num_tasks
+        Map<String, Object> jsonMap = StormCommon.componentConf(component);
+        int numTasks = (int) (long) jsonMap.getOrDefault(Config.TOPOLOGY_TASKS, 0);
```

Comment:
Why do we need to cast twice: `(int) (long) `? Also it's probably better to have 

`jsonMap.getOrDefault(Config.TOPOLOGY_TASKS, numExecutors);`. But in theory, `TOPOLOGY_TASKS` will be set during submission if it's not specified by the topology. So it should be fine.

Reply:
Removed the ugly cast and now using numExecutors as default value